nameresolution/file/file.go

```diff
@@ -0,0 +1,190 @@
+package file
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path/filepath"
+
+	"github.com/gofrs/flock"
+
+	"github.com/dapr/components-contrib/nameresolution"
+	"github.com/dapr/kit/config"
+	"github.com/dapr/kit/logger"
+)
+
+const (
+	dirKey           = "dir"
+	defaultDaprDir   = ".dapr"
+	defaultNamingDir = "naming"
```

Comment:
How about put it into `$HOME/.dapr/components` as it is a component too?

Reply:
Instead of putting it in components dir, it should be configured with $HOME/.dapr/config.yaml (see https://github.com/dapr/components-contrib/pull/1380#issue-1079414460), what do you think, @daixiang0 ?

Reply:
I think so.